packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -567,8 +569,16 @@ List<String> _validateList<T>(Object? yamlList, String context, String typeAlias
     }
   }
 
-  return errors;
+  return errors.isEmpty ? (List<T>.from(yamlList), <String>[]) : (null, errors);
+}
+
+(List<T>? result, List<String> errors) _parseNullableList<T>(Object? yamlList, String context, String typeAlias) {
```

Comment:
do we need this helper, or could we inline this at the callsites? I find the DRY benefits of this kind of trivial abstraction often not worth the cognitive overhead.

Reply:
Addressed in [remove _parseNullableList](https://github.com/flutter/flutter/pull/143557/commits/d1fe0523223cef52c5d9b3a4f3ca32d7fe25833d) and [cleanup](https://github.com/flutter/flutter/pull/143557/commits/7cee3ed79e3a278133342efb8bc3623c47384a84)